egs/gale_mandarin/s5/run.sh

```diff
@@ -130,40 +148,42 @@ steps/align_fmllr.sh --nj $nJobs --cmd "$train_cmd" \
 ## SGMM (subspace gaussian mixture model), excluding the "speaker-dependent weights"
 steps/train_ubm.sh --cmd "$train_cmd" 700 \
  data/train data/lang exp/tri3b_ali exp/ubm5a || exit 1;
- 
+
 steps/train_sgmm2.sh --cmd "$train_cmd" 5000 20000 data/train data/lang exp/tri3b_ali \
   exp/ubm5a/final.ubm exp/sgmm_5a || exit 1;
 
-utils/mkgraph.sh data/lang_dev exp/sgmm_5a exp/sgmm_5a/graph || exit 1;
-
+utils/mkgraph.sh data/lang_test exp/sgmm_5a exp/sgmm_5a/graph || exit 1;
 steps/decode_sgmm2.sh --nj $nDecodeJobs --cmd "$decode_cmd" --config conf/decode.config \
   --transform-dir exp/tri3b/decode exp/sgmm_5a/graph data/dev exp/sgmm_5a/decode &
 
 steps/align_sgmm2.sh --nj $nJobs --cmd "$train_cmd" --transform-dir exp/tri3b_ali \
   --use-graphs true --use-gselect true data/train data/lang exp/sgmm_5a exp/sgmm_5a_ali || exit 1;
 
 ## boosted MMI on SGMM
-steps/make_denlats_sgmm2.sh --nj $nJobs --sub-split 30 --beam 9.0 --lattice-beam 6 \
-  --cmd "$decode_cmd" --transform-dir \
-  exp/tri3b_ali data/train data/lang exp/sgmm_5a_ali exp/sgmm_5a_denlats || exit 1;
-  
+steps/make_denlats_sgmm2.sh --nj $nJobs --sub-split $nJobs --beam 9.0 --lattice-beam 6 \
+  --cmd "$decode_cmd" --num-threads 4 --transform-dir exp/tri3b_ali \
+  data/train data/lang exp/sgmm_5a_ali exp/sgmm_5a_denlats || exit 1;
+
 steps/train_mmi_sgmm2.sh --cmd "$train_cmd" --num-iters 8 --transform-dir exp/tri3b_ali --boost 0.1 \
   data/train data/lang exp/sgmm_5a exp/sgmm_5a_denlats exp/sgmm_5a_mmi_b0.1
- 
+
 #decode GMM MMI
-utils/mkgraph.sh data/lang_dev exp/sgmm_5a_mmi_b0.1 exp/sgmm_5a_mmi_b0.1/graph || exit 1;
+utils/mkgraph.sh data/lang_test exp/sgmm_5a_mmi_b0.1 exp/sgmm_5a_mmi_b0.1/graph || exit 1;
 
 steps/decode_sgmm2.sh --nj $nDecodeJobs --cmd "$decode_cmd" --config conf/decode.config \
-  --transform-dir exp/tri3b/decode exp/sgmm_5a_mmi_b0.1/graph data/dev exp/sgmm_5a_mmi_b0.1/decode &
-  
+  --transform-dir exp/tri3b/decode exp/sgmm_5a_mmi_b0.1/graph data/dev exp/sgmm_5a_mmi_b0.1/decode
+
 for n in 1 2 3 4; do
-  steps/decode_sgmm2_rescore.sh --cmd "$decode_cmd" --iter $n --transform-dir exp/tri3b/decode data/lang_dev \
+  steps/decode_sgmm2_rescore.sh --cmd "$decode_cmd" --iter $n --transform-dir exp/tri3b/decode data/lang_test \
     data/dev exp/sgmm_5a_mmi_b0.1/decode exp/sgmm_5a_mmi_b0.1/decode$n
-  
-  steps/decode_sgmm_rescore.sh --cmd "$decode_cmd" --iter $n --transform-dir exp/tri3b/decode data/lang_dev \
+done
+
+for n in 1 2 3 4; do
+  steps/decode_sgmm2_rescore.sh --cmd "$decode_cmd" --iter $n --transform-dir exp/tri3b/decode data/lang_test \
     data/dev exp/sgmm_5a/decode exp/sgmm_5a_mmi_onlyRescoreb0.1/decode$n
 done
 
+wait
 local/nnet/run_dnn.sh
 
```

Comment:
in local/nnet/run_dnn.sh, there is a invalid option to steps/nnet/train.sh (--use-gpu-id) and a missing `done` near the end of the script.